e2e/tests/app_stake_transfer.feature

```diff
@@ -0,0 +1,18 @@
+Feature: App Stake Transfer Namespace
+
+    Scenario: User can transfer Application stake to non-existing application address
+        Given the user has the pocketd binary installed
+        And this test ensures the "application" for account "app2" is not staked
+        And this test ensures the "application" for account "app3" is not staked
+        # Stake with 1 uPOKT more than the current stake used in genesis to make
+        # the transaction succeed.
+        And the account "app2" has a balance greater than "1000070" uPOKT
+        And an account exists for "app3"
+        And the user successfully stakes a "application" with "1000070" uPOKT for "anvil" service from the account "app2"
+        When the user transfers the "application" stake from account "app2" to account "app3"
+        Then the user should be able to see standard output containing "txhash:"
+        And the user should be able to see standard output containing "code: 0"
```

Comment:
I'm confused.

1. I noticed we're not using `TheUserStakesAWithUpoktFromTheAccount`, which could be good here
2. The new `TheUserSuccesfully...` does some of the checks that are done explicitly in this feature too

We need to either:
1. Reuse existing helpers
2. Remove the extra lines

Reply:
> I'm confused.
> 
> 1. I noticed we're not using `TheUserStakesAWithUpoktFromTheAccount`, which could be good here

Are you saying that that would be sufficient and that `...WithService...` is excessive?

> 2. The new `TheUserSuccesfully...` does some of the checks that are done explicitly in this feature too

The assertions regarding the output containing `"txhash:"` and `"code: 0"` MUST be duplicated because they assert on the output of the most recently run command (i.e. the step prior). We could create a new step that combines these two which would be something like `TheUserShouldNotSeeACLIError` but would still need to call it in both places.

Reply:
UPDATE: I'm just now noticing that `ThePocketBinaryShouldExitWithoutError()` is already a thing. Wdyt about consolidating the output assertion steps into, and replacing usages of the consolidated steps with, it?

Reply:
> 1. I noticed we're not using `TheUserStakesAWithUpoktFromTheAccount`, which could be good here

UPDATE: Replacing usage of `TheUserStakesAWithUpoktForServiceFromTheAccount` with `TheUserStakesAWithUpoktFromTheAccount` seems to produce a new error:

```
=== RUN   TestFeatures
=== RUN   TestFeatures/App_Stake_Transfer_Namespace
=== RUN   TestFeatures/App_Stake_Transfer_Namespace/User_can_transfer_Application_stake_to_non-existing_application_address
    init_test.go:157: Step: the user has the pocketd binary installed
    init_test.go:157: Step: this test ensures the "application" for account "app2" is not staked
    init_test.go:157: Step: this test ensures the "application" for account "app3" is not staked
    init_test.go:157: Step: the account "app2" has a balance greater than "1000070" uPOKT
    init_test.go:157: Step: an account exists for "app3"
    init_test.go:157: Step: the user successfully stakes a "application" with "1000070" uPOKT from the account "app2"
    init_test.go:177: 
                Error Trace:    /home/bwhite/Projects/pokt/poktroll/e2e/tests/init_test.go:177
                                                        /home/bwhite/Projects/pokt/poktroll/e2e/tests/stake_app_transfer_steps_test.go:13
                                                        /home/bwhite/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.2.linux-amd64/src/reflect/value.go:596
                                                        /home/bwhite/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.2.linux-amd64/src/reflect/value.go:380
                                                        /home/bwhite/go/pkg/mod/github.com/regen-network/gocuke@v1.1.0/run_step.go:103
                                                        /home/bwhite/go/pkg/mod/github.com/regen-network/gocuke@v1.1.0/run_scenario.go:121
                                                        /home/bwhite/go/pkg/mod/github.com/regen-network/gocuke@v1.1.0/run_scenario.go:70
                                                        /home/bwhite/go/pkg/mod/github.com/regen-network/gocuke@v1.1.0/run_doc.go:34
                Error:          "" does not contain "txhash:"
--- FAIL: TestFeatures (2.37s)
    --- FAIL: TestFeatures/App_Stake_Transfer_Namespace (2.37s)
        --- FAIL: TestFeatures/App_Stake_Transfer_Namespace/User_can_transfer_Application_stake_to_non-existing_application_address (2.37s)
FAIL
FAIL    github.com/pokt-network/poktroll/e2e/tests      2.457s
FAIL
```

Reply:
I'm not really following the question / ask at a high level.

Overall:
1. If it's easier to just keep the other step -> let's do it and move on.
2. If there seems to be an underlying issue with `TheUserStakesAWithUpoktFromTheAccount`, lmk and I'll pull the branch to debug/log/etc myself.

Lmk if you prefer (1) or (2).

Reply:
> I noticed we're not using TheUserStakesAWithUpoktFromTheAccount, which could be good here

There's no issue with the step as it is. I was trying to understand your suggestion and my best interpretation was that you were suggesting swapping the step out for the other.

> If it's easier to just keep the other step -> let's do it and move on.

:+1: